### ISSUE TITLE
Claim task (FE)

### ISSUE THREAD

**Issue by MariusJulius** [keeleinstituut/tv-tolkevarav#568]

**Comment by plakitkelly** [comment]
Sending task to vendors, task should appear in Ootel ülesanded. Currently it appears, but these tasks are visible for these users, who are not assigned to task. 
![image](https://github.com/keeleinstituut/tv-tolkevarav/assets/130659408/50bb28a5-a273-41ee-ad09-c4f206d4552a)
This user is not assigned to any task but I can see in this list all tasks that have beed forwarded to vendors.

Tab titled "Ootel ülesanded" - vendors can see the tasks where they have been added to the task's candidate group
4.2.2 System displays the tab where is the list of tasks the user can claim

I understand that I shouldn't see these tasks in the list. And if I try to accept any task, I see error: "The vendor is not a candidate for the task"

**Comment by KaarelKa** [comment]
@MariusJulius BE issue, I think @kadmit is aware of this as well, and I'm guessing we already have some task for it. 
From what I can see there can be 2 cases that result in this issue:

1. A Vendor is removed from vendors list after he has been added to some assignments as a candidate
2. Users with VIEW_VENDOR_TASK privilege can see tasks, which are not their own. As far as I know, they can only see tasks assigned to other people, but maybe they can also see unassigned tasks, which are not assigned to them ?

**Comment by plakitkelly** [comment]
@KaarelKa It's still BE issue?

**Comment by KaarelKa** [comment]
@plakitkelly Should be yeah, can you give me the PIC of the user you used for testing ? I'll take a quick look. But most likely @thenouan needs to check it out

**Comment by plakitkelly** [comment]
@KaarelKa 37808106687
You can check orders with ending 241 (vendor) and 242 (not vendor).

**Comment by KaarelKa** [comment]
@thenouan Yeah, looks like a BE issue. Fetching unassigned tasks also returns tasks, where the current user is not a candidate.

curl:

curl 'https://api.dev.tolkevarav.eki.ee/translation-order/api/workflow/tasks?page=1&per_page=10&assigned_to_me=0' \
  -H 'authority: api.dev.tolkevarav.eki.ee' \
  -H 'accept: application/json, text/plain, */*' \
  -H 'accept-language: en-US,en;q=0.9,ceb;q=0.8' \
  -H 'authorization: Bearer eyJhbGciOiJSUzI1NiIsInR5cCIgOiAiSldUIiwia2lkIiA6ICJBRzRYOXR2SVhubkoxWUhiYnJHZmdYdnNYNEVVU2xGalk0RkZSekY5MjlRIn0.eyJleHAiOjE3MDQyODI5MjIsImlhdCI6MTcwNDI4MjAyMiwiYXV0aF90aW1lIjoxNzA0MjgxNTc2LCJqdGkiOiJjNjkyZGQ2Yy1iNjY4LTQ2MjQtOGM1Mi04NTA1NjNhZjdhZTAiLCJpc3MiOiJodHRwczovL3Nzby5kZXYudG9sa2V2YXJhdi5la2kuZWUvcmVhbG1zL3RvbGtldmFyYXYtZGV2IiwiYXVkIjoiYWNjb3VudCIsInN1YiI6IjU2MzgwNmExLTY0YjMtNDc5Zi04ZDY4LTdkZDQzNGYxN2Q4OCIsInR5cCI6IkJlYXJlciIsImF6cCI6IndlYiIsIm5vbmNlIjoiMmNmYzgxNmQtOGM5Ny00MGY1LThkMTYtOGUyYjNkNmM0ZWQ4Iiwic2Vzc2lvbl9zdGF0ZSI6ImJjMDJjMzk5LWEwZDUtNDFmZC1iOTk3LTVkYjNlNmVkYzVmYSIsImFjciI6IjAiLCJhbGxvd2VkLW9yaWdpbnMiOlsiaHR0cHM6Ly9kZXYudG9sa2V2YXJhdi5la2kuZWUiLCJodHRwOi8vbG9jYWxob3N0OjMwMDAiXSwicmVhbG1fYWNjZXNzIjp7InJvbGVzIjpbIm9mZmxpbmVfYWNjZXNzIiwiZGVmYXVsdC1yb2xlcy10b2xrZXZhcmF2LWRldiIsInVtYV9hdXRob3JpemF0aW9uIl19LCJyZXNvdXJjZV9hY2Nlc3MiOnsiYWNjb3VudCI6eyJyb2xlcyI6WyJtYW5hZ2UtYWNjb3VudCIsIm1hbmFnZS1hY2NvdW50LWxpbmtzIiwidmlldy1wcm9maWxlIl19fSwic2NvcGUiOiJvcGVuaWQgZW1haWwgcHJvZmlsZSIsInNpZCI6ImJjMDJjMzk5LWEwZDUtNDFmZC1iOTk3LTVkYjNlNmVkYzVmYSIsInRvbGtldmFyYXYiOnsicGVyc29uYWxJZGVudGlmaWNhdGlvbkNvZGUiOiIzNzgwODEwNjY4NyIsInVzZXJJZCI6IjlhNTNhODBiLTY2ODAtNGVjMC05ODU2LTk4N2Y3NGYwMWIwOCIsImluc3RpdHV0aW9uVXNlcklkIjoiOWE1M2E4MGItNjZmNC00MzA3LWFlYjYtNGI3ZjQzYzIxOTUzIiwiZm9yZW5hbWUiOiJPYXNpaiIsInN1cm5hbWUiOiJDaWphIiwic2VsZWN0ZWRJbnN0aXR1dGlvbiI6eyJpZCI6IjlhMjE4MmZhLTcwZWEtNDcyNy1hMjU0LTU3MmFjZTM1Nzk1NiIsIm5hbWUiOiJUZXN0aW1pbmVRIFRlc3RpbWluZVEgVGVzdGltaW5lUSBUZXN0aW1pbmVRIFRlc3RpbWluZVEgVGVzdGltaW5lUSBUZXN0aW1pbmVRIFRlc3RpbWluZVEgVGVzdGltaW5lUSJ9LCJwcml2aWxlZ2VzIjpbIlZJRVdfVkVORE9SX0RCIl19LCJlbWFpbF92ZXJpZmllZCI6ZmFsc2UsIm5hbWUiOiJlZTM3ODA4MTA2Njg3LWZpcnN0LW5hbWUgZWUzNzgwODEwNjY4Ny1sYXN0LW5hbWUiLCJwcmVmZXJyZWRfdXNlcm5hbWUiOiJlZTM3ODA4MTA2Njg3IiwiZ2l2ZW5fbmFtZSI6ImVlMzc4MDgxMDY2ODctZmlyc3QtbmFtZSIsImZhbWlseV9uYW1lIjoiZWUzNzgwODEwNjY4Ny1sYXN0LW5hbWUifQ.gSg2hJKH3BEEFE7XoQ8YhmWZCPCGgJG8Z7nQ7700IEJEx0sUvYp-J_tPw2rNLu5YdDnk_NxO-g2fuH6vL8sedIC8rP5z8eqMwyBRsyZO5Hy5m0gClgQ50Fhq_is_OYE8nFI83hK5Xwr7E74pGcNKnb856uLyxjmbjzrNuSwJBsBLK-KIBZWab6lzYkBaDGywEGfn0McXBrUf2-bcqs-50Z5FX5a2vb0SPmDkdydsrTzZG3hgFTt1b7fYqzFXn7a7bKaPPm2VaI7RiIwKV7bMhh4K537FyWRipy25EiTHyA48Lsxaq0d5kYEF5ZNg2l5ov2nELkZJGBfFHsEzjzJBhQ' \
  -H 'cache-control: no-cache' \
  -H 'origin: http://localhost:3000' \
  -H 'pragma: no-cache' \
  -H 'referer: http://localhost:3000/' \
  -H 'sec-ch-ua: "Not_A Brand";v="8", "Chromium";v="120", "Google Chrome";v="120"' \
  -H 'sec-ch-ua-mobile: ?0' \
  -H 'sec-ch-ua-platform: "macOS"' \
  -H 'sec-fetch-dest: empty' \
  -H 'sec-fetch-mode: cors' \
  -H 'sec-fetch-site: cross-site' \
  -H 'user-agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/120.0.0.0 Safari/537.36' \
  --compressed

**Comment by thenouan** [comment]
@KaarelKa this is handled on BE side actually. FE needs to pass is_candidate=1 as queryparam additionally

**Comment by kadmit** [comment]
@KaarelKa to fetch tasks where the active user is a candidate please use the `is_candidate=1&assigned_to_me=0` query params. Please note that tasks with the types `REVIEW` and `CLIENT_REVIEW` don't have candidates, they only have assignees.

**Comment by MariusJulius** [comment]
This seems still not working people with task "We were concerned that the sworn translator would see all the assigned tasks, not only the work submitted to him - just yesterday the situation was the same in the dev environment" 

@KaarelKa I understood it should be fixed after last deploy?

**Comment by plakitkelly** [comment]
Tested in 07.03 - It's OK
